### PR TITLE
Checklist

### PR DESCRIPTION
**Pull request by ichikareman2**
@justemoilouise

please review implementation :D

### CODE REVIEW

File: components/checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { HTMLAttributes, InputHTMLAttributes } from 'react'
+export type CheckboxProps = HTMLAttributes<HTMLInputElement> & InputHTMLAttributes<HTMLInputElement>
+export default function Checkbox(props: CheckboxProps) {
+  return (
+    <div>
```
Comment (justemoilouise) [inline components/checkbox.tsx]: nit: remove extra `div` if won't be used

File: components/checklist/checklistInput.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  onCancel?: () => void
+}
+
+export default function ChecklistInput({ onSubmit, value, onChange, onCancel }: ChecklistInputProps) {
+  const handleKeydown = (e: KeyboardEvent) => {
+    if (e.key === 'Enter') { onSubmit() }
+    else if (e.key === 'Escape') { onCancel?.() }
+  }
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => onChange(e.currentTarget.value)
+  return (<>
```
Comment (justemoilouise) [inline components/checklist/checklistInput.tsx]: nit: remove extra `<>`

File: components/checklist/checkItem.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import { useState } from 'react';
+import ChecklistInput from './checklistInput';
+
+export type CheckItemProps = {
+  item: ChecklistItem;
+  onUpdate: (item: ChecklistItem) => void
+}
+export default function CheckItem({item, onUpdate}: CheckItemProps) {
+  const [editable, setEditable] = useState(false)
+  const [tempDescription, setTempDescription] = useState(item.description)
```
Comment (justemoilouise) [inline components/checklist/checkItem.tsx]: nit: add state object types

File: components/checklist/checklist.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    const checklistItemDesc = newItemValue
+    if (!checklistItemDesc) { return }
+    const newItem = createChecklistItem(checklistItemDesc)
+    setChecklist([...checklist, newItem])
+    setNewItemValue('')
+  }
+  const updateChecklist = (checklistItem: ChecklistItem) => {
+    const updateIndex = findIndex<ChecklistItem>(idEqualto(checklistItem.id), checklist)
+    const newList = update(updateIndex, checklistItem, checklist)
+    if(checklistItem) { setChecklist(newList)}
```
Comment (justemoilouise) [inline components/checklist/checklist.tsx]: isn't this always called since `checklistItem` is truthy?

File: components/checklist/checkItem.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  const handleClick = () => {
+    setTempDescription(item.description)
+    setEditable(!editable)
+  }
+
+  const handleCancel = () => {
+    setTempDescription(item.description)
+    setEditable(false)
```
Comment (justemoilouise) [inline components/checklist/checkItem.tsx]: duplicated with `updateDesc`, perhaps can be further simplified

**Review by justemoilouise — APPROVED** [review]
LGTM! with nits and questions, but up to you if you want to update ✌🏼

File: components/checklist/checkItem.tsx (hunk 5 above)
Comment (ichikareman2) [reply components/checklist/checkItem.tsx]: wachumean? please elaborate.

**Comment by ichikareman2** [comment]
@justemoilouise please check fix for on click not focusing on html input element
files on checklistInput - for forwarding ref so that focus() can be callsed
and checkItem - added effect hook to call focus when `editable` state changes

**Review by justemoilouise — APPROVED** [review]
charesso! 👍🏼

**Pull request merged**